### PR TITLE
Models-per-category changes

### PR DESCRIPTION
**Pull request by akashenfelter**
The changes suggested by Oscar to improve the script before making it public in the gallery

### CODE REVIEW

File: model-per-category/create-category-models/script.whizzml [hunk 1]
```diff
@@ -27,7 +27,10 @@
 ;; that have the given category.
 (define (category-dataset origin-id field category)
   (let (fltr (flatline "(= (f {{field}}) {{category}})"))
-    (create-dataset origin-id {"lisp_filter" fltr})))
+    (create-dataset origin-id {"lisp_filter" fltr
+                               "name" (str ((fetch origin-id) "name")
```
Comment (jaor) [inline model-per-category/create-category-models/script.whizzml]: this function is called for all categories, always with the same dataset, so re-fetching it all the time is  a bit inefficient: we can make this function take a name argument and compute that name only once just before we loop over `category-dataset` below...

**Review by jaor — CHANGES_REQUESTED** [review]
an optimization...

**Comment by akashenfelter** [comment]
Sorry this fell off my radar. Nit picked!

File: model-per-category/create-category-models/script.whizzml [hunk 2]
```diff
@@ -40,7 +41,8 @@
 ;; Final workflow for multiway first split
 (define (multiway-model dataset field objective kind parameters)
   (let ([cs missings id name] (categories dataset field)
-        ds (for (c cs) (category-dataset dataset field c))
+        name ((fetch dataset) "name")
```
Comment (jaor) [inline model-per-category/create-category-models/script.whizzml]: i don't think that's right: your `name` here is shadowing the `name` returned by `(categories ..)`

**Review by jaor — CHANGES_REQUESTED** [review]
i think there's a problem...

**Comment by akashenfelter** [comment]
Better?

**Comment by whizzmler** [comment]
no, i'm afraid not :) you changed the name at the wrong place, it's in
the let before calling category-dataset that it must be changed... we
had:

```clojure
(let ([cs missings id name] (categories dataset field)
      ps ...
  <code that, presumable uses `name` above>))
```

and then you added

```clojure
(let ([cs missings id name] (categories dataset field)
      name ((fetch dataset) "name") ;; <-- this name is shadowing the name in pthe previous line
      ps ...
  <code that, presumable uses `name` above, but now `name` is different>))
```

(the name of the parameter can be `name`, there's no shading there).


-- 
No, no, you're not thinking, you're just being logical.
 -Niels Bohr, physicist (1885-1962)

**Comment by akashenfelter** [comment]
Sorry, I wasn't thinking.

**Pull request merged**